cpp/src/datetime/datetime_ops.cu

```diff
@@ -180,6 +192,109 @@ std::unique_ptr<column> apply_datetime_op(column_view const& column,
   return output;
 }
 
+struct add_months_functor {
+  column_view timestamp_column;
+  column_view months_column;
+  mutable_column_view output;
+
+  add_months_functor(column_view tsc, column_view mc, mutable_column_view out)
+    : timestamp_column(tsc), months_column(mc), output(out)
+  {
+  }
+
+  // std chrono implementation is copied here due to nvcc bug
+  // https://howardhinnant.github.io/date_algorithms.html#days_from_civil
+  static CUDA_DEVICE_CALLABLE timestamp_D
```

Comment:
What bug? Also, you can't use other software without a proper license and SWIPAT.

Reply:
Howard's [`date`](https://github.com/HowardHinnant/date/blob/master/LICENSE.txt) lib is MIT, which is compatible with Apache 2, but we need a SWIPAT.

Reply:
>> What bug?

it is [this](https://github.com/rapidsai/thirdparty-libcxx/pull/4) issue.

as i mentioned in the comment, the `std::chrono` implementation is copied here with some modifications to workaround the issue mentioned above. the reference link to algorithm was also copied from the `std::chrono` code.

Reply:
the original implementation reference link is [this](https://github.com/rapidsai/thirdparty-libcxx/blob/rapidsai-interop/include/chrono#L2379)

Reply:
moved to #5931